LAPACK: add prebuilt dll for MSVC

NeroBurner opened the pull request:
add lapack 3.7.1 prebuilt for MSVC amd64

This adds a prebuilt dll hosted at https://github.com/NeroBurner/lapack-prebuilt

on MSVC I wasn't able to add a Fortran compiler. But for Unicode support I need to build with MSVC (I wasn't able to make boost::filesystem and MINGW behave).
My solution is to download the prebuilt c-library and provide a cmake-config file when LAPACK is requested when using MSVC

currently I only provide amd64 architecture

ruslo commented on cmake/projects/LAPACK/hunter-msvc.cmake:
```diff
@@ -0,0 +1,32 @@
+    VERSION
+    "3.7.1"
+    URL
+    https://github.com/NeroBurner/lapack-prebuilt/archive/v3.7.1-msvc-amd64.zip
```
I've saved it to hunter-packages/lapack, please use this link:
* https://github.com/hunter-packages/lapack/releases/download/v3.7.1-p0/lapack-prebuilt-3.7.1-msvc-amd64.zip

ruslo commented on cmake/projects/LAPACK/hunter-msvc.cmake:
```diff
@@ -0,0 +1,32 @@
+
+# determine architecture
+include(hunter_check_toolchain_definition)
+hunter_check_toolchain_definition(NAME "_WIN64"    DEFINED WIN64_defined)
```
Please use `_WIN64_defined` or `_hunter_win64_defined` here.

ruslo replied:
Add `hunter_report_broken_package` if `_WIN64` is not defined, example:
* https://github.com/ruslo/hunter/blob/d7c2b1da4423077c46fd1e89bd1e25120666ed78/cmake/projects/CLAPACK/hunter.cmake#L13

NeroBurner commented:
updated

> >    on MSVC I wasn't able to add a Fortran compiler. But for Unicode support I need to build with MSVC (I wasn't able to make boost::filesystem and MINGW behave).
> 
> Not sure what it means. You have to use Fortran from MinGW to build LAPACK with MSVC toolchain?

When compiling a project with MSVC which requests the package LAPACK, instead of downloading the source and trying to compiling the Fortran code I let hunter download the prebuilt C-dll.

The dll and lib files are from http://icl.cs.utk.edu/lapack-for-windows/lapack/index.html#libraries.
The cmake-config files are from the install step when compiling with mingw-w64, and I updated them manually to link to the prebuilt dll and lib files.

ruslo commented:
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2351
* https://travis-ci.org/ingenue/hunter/builds/272937751?utm_source=github_status&utm_medium=notification

ruslo commented:
> instead of downloading the source and trying to compiling the Fortran code

That's kind of what I was really asking. Why we can't compile this from sources?

NeroBurner commented:
compiling from source would be a mixed toolchain. For LAPACK we need mingw-w64-gfortran, for the rest of the project MSVC is used. I don't know how to do that

ruslo commented:
> compiling from source would be a mixed toolchain. For LAPACK we need mingw-w64-gfortran, for the rest of the project MSVC is used

Okay, that's what I was thinking about. One more use case for this feature request: https://github.com/ruslo/hunter/issues/495

NeroBurner commented:
Shall I add a pr to add MSVC tests?
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

ruslo commented:
> Shall I add a pr to add MSVC tests?

After tests passed I will merge it to master and update `pkg.lapack`. Then you can send pull request with enabling MSVC toolchains.

ruslo commented:
> Testing

https://github.com/ruslo/hunter/issues/1021

ruslo commented:
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2353
* https://travis-ci.org/ingenue/hunter/builds/274517349?utm_source=github_status&utm_medium=notification

ruslo commented:
Fix .travis.yml, restart tests:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2356
* https://travis-ci.org/ingenue/hunter/builds/274529574?utm_source=github_status&utm_medium=notification

ruslo commented:
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2358
* https://travis-ci.org/ingenue/hunter/builds/274562311?utm_source=github_status&utm_medium=notification

ruslo commented:
Released: https://github.com/ruslo/hunter/releases/tag/v0.19.98

NeroBurner commented:
could you push your local `pkg.lapack` branch. I see the "Merge branch 'pk.template' into pkg.lapack" commit message, but only in the branch `upload.lapack`. The branch `pkg.lapack` is a few commit behind

ruslo commented:
> could you push your local pkg.lapack branch

Done
